.github/workflows/deploy.yml

```diff
@@ -1,68 +1,81 @@
-        uses: peter-evans/create-pull-request@5e914681df9dc83aa4e4905692ca88beb2f9e91f # v7.0.5
-        with:
-          token: ${{ secrets.SGTPOOKI_PAT }}
-          base: staging
-          branch: staging-release
-          title: 'chore: point staging to release ${{ steps.get_latest_tag.outputs.tag }}'
-          body: 'This PR points the staging branch to release ${{ steps.get_latest_tag.outputs.tag }}.  **Choose "Rebase and merge" for this PR only!**'
-
-  # from https://github.com/peter-evans/create-pull-request/blob/main/docs/examples.md#keep-a-branch-up-to-date-with-another
-  deploy-production:
-    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'push' && github.ref == 'refs/heads/staging')
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          token: ${{ secrets.SGTPOOKI_PAT }}
-          ref: production
-          fetch-depth: 0  # Fetch all branches and commit history
-      - name: Get latest tag deployed to staging
-        id: get_latest_tag
-        run: |
-          latest_tag=$(git --no-pager log origin/staging --all --grep="chore: point staging to release v" --pretty=format:"%H %s" | head -n1 | sed -n 's/.*release \(v[0-9]*\.[0-9]*\.[0-9]*\).*/\1/p')
-          echo "tag=$latest_tag" >> $GITHUB_OUTPUT
-      - name: Set production branch to latest tag deployed to staging
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+      - name: Verify tag
+        env:
+          TAG: ${{ inputs.tag }}
+          ENVIRONMENT: ${{ inputs.environment }}
         run: |
-          git reset --hard ${{ steps.get_latest_tag.outputs.tag }}
-      - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v7.0.5
-        with:
-          token: ${{ secrets.SGTPOOKI_PAT }}
-          base: production
-          branch: production-release
-          title: 'chore: Point production to ${{ steps.get_latest_tag.outputs.tag }}'
-          body: 'This PR points the production branch to release ${{ steps.get_latest_tag.outputs.tag }}, which is the latest tag deployed to staging. **Choose "Rebase and merge" for this PR only!**'
-
+          # If deploying to staging, ensure the tag is somewhere in the history of master
+          if [ "$ENVIRONMENT" == "staging" ]; then
+            git merge-base --is-ancestor $TAG origin/master
+            if [ $? -ne 0 ]; then
+              echo "Tag $TAG is not present in the history of master"
+              exit 1
+            fi
+          fi
+          # If deploying to production, ensure the tag is present in the history of staging
+          if [ "$ENVIRONMENT" == "production" ]; then
+            git merge-base --is-ancestor $TAG origin/staging
+            if [ $? -ne 0 ]; then
+              echo "Tag $TAG is not present in the history of staging"
+              exit 1
+            fi
+          fi
```

Comment:
nit: iiuc staging will be same rule as prod, tag should be attached to commit from main branch

Reply:
Got it, thanks. Makes things easier :)